keras_core/ops/numpy.py

```diff
@@ -1534,6 +1535,38 @@ def diagonal(x, offset=0, axis1=0, axis2=1):
     )
 
 
+class Digitize(Operation):
+    def call(self, x, bins):
+        return backend.numpy.digitize(x, bins)
+
+    def compute_output_spec(self, x, bins):
+        return KerasTensor(x.shape, dtype=x.dtype)
```

Comment:
Surely the dtype should be int? What is it?

Reply:
`x.dtype` can be float or int, so I must return `KerasTensor(x.shape, dtype="int32")`? 
actually I wanted to make the dtype int, but I saw that there is other ops where the input can be float and the dtype is set to the input dtype. so I thought the dtype of the shapemust be the same as the input.

Reply:
Ok I get it now. it's `int64` for numpy.

Reply:
It must be the same dtype in all backends (if it isn't, that's a bug and we need to cast)

Reply:
I have used `standardize_dtype` to test the return dtypes but it changes the dtype of numpy from `int64` to `int32` because of this line https://github.com/keras-team/keras-core/blob/6b4d0549ea0402114f92a1e668fc12406b9cf023/keras_core/backend/common/variables.py#L391

where:
```python
np.dtype("int64") == "int" ##returns True
```

so it executes what in the condition and return `int32` instead of `int64` which causes the fail of the test. I can open a pull request to fix this. I will just move the `if` statement below it to the top.

Reply:
also jax doesn't enable `int64` until `jax_enable_x64` is set to `True` using `jax.config.update("jax_enable_x64", True)`. I think If we want to unify the dtypes we should enable x64 it when the backend is set to jax.

Reply:
Got it. In this case just pass `dtype="int"` in `compute_output_spec`, and in the unit tests check `x.dtype` to match `backend.standardize_dtype("int")`.

Reply:
I did that. but I had to cast pytorch output to `int32` for the test to work. 
I think fixing fixing `standrize_dtype()` and enabling `int64` by default for jax will be a better solution. can you give that a look?

Reply:
JAX doesn't make that possible.